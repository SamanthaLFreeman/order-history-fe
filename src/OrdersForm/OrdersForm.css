form {
  display: flex;
  align-content: space-around;
  flex-direction: column;
  height: 100%;
  width: 300px;
}

input {
  border: none;
  margin: 15px 0;
  padding: 8px;
}

button {
  border: none;
  background-color: green;
  color: #fff;
  font-weight: bold;
  margin: 15px 0;
  padding: 10px;
}
