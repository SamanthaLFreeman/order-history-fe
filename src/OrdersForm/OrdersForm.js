import React, { Component } from 'react';
import './OrdersForm.css';

class OrdersForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      img: '',
      description: '',
      price: ''
    }
  }

  render() {
    return (
      <form>
        <input placeholder="Image Url:" />
        <input placeholder="Name:" />
        <input placeholder="Description:" />
        <input placeholder="Price:" />
        <button>Add Purchase:</button>
      </form>
    )
  }
}

export default OrdersForm;