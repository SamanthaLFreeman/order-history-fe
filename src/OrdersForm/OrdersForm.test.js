import React from 'react';
import { shallow } from 'enzyme';
import OrdersForm from './OrdersForm';

describe('OrdersForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<OrdersForm addOrder={jest.fn()} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update the state when handleChange is callled', () => {
    const mockEvent = {target: {name: 'name', value: 'jeans'}}
    const expected = 'jeans'

    wrapper.instance().handleChange(mockEvent);
    
    expect(wrapper.state('name')).toEqual(expected)
  })

  it('should run handleSubmit when the button is clicked', () => {
    wrapper.instance().handleSubmit = jest.fn();
    const mockEvent = {preventDefault: jest.fn()};

    wrapper.instance().forceUpdate();
    wrapper.find('button').simulate('click', mockEvent);

    expect(wrapper.instance().handleSubmit).toHaveBeenCalled()
  })

});