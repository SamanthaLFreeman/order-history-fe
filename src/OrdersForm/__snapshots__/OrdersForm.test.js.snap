// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersForm should match the snapshot 1`] = `
<form>
  <input
    name="img"
    onChange={[Function]}
    placeholder="Image Url:"
    type="text"
    value=""
  />
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name:"
    type="text"
    value=""
  />
  <input
    name="description"
    onChange={[Function]}
    placeholder="Description:"
    type="text"
    value=""
  />
  <input
    name="price"
    onChange={[Function]}
    placeholder="Price:"
    type="number"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add Purchase:
  </button>
</form>
`;
